src/typeahead/typeahead.js

```diff
@@ -399,13 +399,30 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
     };
   }])
 
-  .filter('typeaheadHighlight', function() {
+  .filter('typeaheadHighlight', ['$sce', '$injector', '$log', function($sce, $injector, $log) {
+
+    var $sanitize;
+    try {
+      $sanitize = $injector.get('$sanitize'); //tries to inject the sanitize service
+    }
+    catch(e) {
+      $log.warn('Unsafe use of typeahead please use ngSanitize'); // Warn the user about the danger
+    }
 
     function escapeRegexp(queryToEscape) {
-      return queryToEscape.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1');
+      return queryToEscape.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1'); // Regex: capture the whole query string and replace it with the string that will be used to match the results, for example if the capture is "a" the result will be \a
     }
 
     return function(matchItem, query) {
-      return query ? ('' + matchItem).replace(new RegExp(escapeRegexp(query), 'gi'), '<strong>$&</strong>') : matchItem;
+      if($sanitize) { // Is the $sanitize service available?
+        if(/<.*>/g.test(matchItem)){
```

Comment:
I think this check isn't required. It's safe to $sanitize something even if it doesn't contain any HTML tags.